### PR TITLE
[4.0] Namespace com_categories

### PR DESCRIPTION
**Pull request by joomdonation**
### Summary of Changes
This simple PR convert com_categories component to namespace MVC for Joomla 4.

### Testing Instructions
Overall, test to make sure it works in the same as before (none namespace version)

1. Install latest Joomla 4.0-dev with sample data (or better install my branch https://github.com/joomdonation/joomla-cms/archive/namespace_com_categories.zip)

2. Login to administrator area of the site, access to Content -> Categories.

3. Try to add/edit/publish/unpublish/archieve categories, make sure it is working as expected

### Additional comment
com_categories require different access permission check compare to most backend component. I ignore the check for now and wait for decision about the method I propose on PR https://github.com/joomla/joomla-cms/pull/15812 . If it is accepted, I will implement the check here or on a separate PR.

### CODE REVIEW

File: administrator/components/com_categories/Controller/Categories.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+class Categories extends Admin
+{
+	/**
+	 * Proxy for getModel
+	 *
+	 * @param   string  $name    The model name. Optional.
+	 * @param   string  $prefix  The class prefix. Optional.
+	 * @param   array   $config  The array of possible config values. Optional.
+	 *
+	 * @return  \Joomla\Cms\Model\Model  The model.
```
Comment (brianteeman) [inline administrator/components/com_categories/Controller/Categories.php]: Shoudnt this be CMS

File: administrator/components/com_categories/Controller/Category.php [hunk 2]
```diff
@@ -164,14 +170,14 @@ protected function getRedirectToListAppend()
 	/**
 	 * Function that allows child controller access to model data after the data has been saved.
 	 *
-	 * @param   JModelLegacy  $model      The data model object.
-	 * @param   array         $validData  The validated data.
+	 * @param   \Joomla\Cms\Model\Model  $model      The data model object.
```
Comment (brianteeman) [inline administrator/components/com_categories/Controller/Category.php]: CMS as before and probably elsewhere

**Comment by brianteeman** [comment]
@wilsonge were my comments not valid?

**Comment by wilsonge** [comment]
They are - but they are also valid in the other PR's I already merged before I got to this one. I'm just about to go do a global search replace :)

**Comment by brianteeman** [comment]
i can do it if you want

**Comment by wilsonge** [comment]
Sure feel free!

**Comment by brianteeman** [comment]
see #15860

**Pull request merged**